core/embed/trezorhal/entropy.c

```diff
@@ -52,3 +56,19 @@ void collect_hw_entropy(void) {
                         FLASH_OTP_BLOCK_SIZE),
          NULL);
 }
+
+void add_sw_entropy(uint8_t *data, size_t data_length) {
+  size_t sw_entropy_index_local = sw_entropy_index % sizeof(SW_ENTROPY_DATA);
+
+  SW_ENTROPY_DATA[sw_entropy_index_local] ^= uwTick & 0xff;
```

Comment:
I see that we automatically reseed from the entropy pool every 1 second, so this means that the amount of entropy that the second byte would contribute is around 2 bits, which isn't all that much ...

Reply:
I changed it to mix all four bytes. From the entropy point of view it can't be worse.

Fixed in https://github.com/trezor/trezor-firmware/commit/0c69dcaaf4a13afd3d6d35e9d423dc9932e926b6.